appstudio-controller/controllers/appstudio.redhat.com/snapshotenvironmentbinding_controller.go

```diff
@@ -178,6 +190,10 @@ func (r *SnapshotEnvironmentBindingReconciler) Reconcile(ctx context.Context, re
 
 	// Update the status field with statusField vars (even if an error occurred)
 	binding.Status.GitOpsDeployments = statusField
+	if err := addComponentDeploymentCondition(binding, ctx, rClient, log); err != nil {
```

Comment:
I think in case of error also function will set condition to true {status: True, reason: CommitsSynced} instead of False {status: False, reason: ErrorOccurred}, as it is not passing the type of condition needs to set, right?

Reply:
You are correct that there is a problem in `addComponentDeploymentCondition` if that function encounters an error, it may set an condition on the binding with all fields at their default (which will be status set to false and reason set to the empty string); it should not do that.  I will fix this.